### DISCUSSION TITLE
StorageApiError: new row violates row-level security policy - bucket creation client

### DISCUSSION THREAD

**Discussion by danielxpander** [supabase/supabase#23095] [Questions] [answered]
I get StorageApiError: new row violates row-level security policy
When I try supabaseClient.storage.createBucket("bucket-name", {
      public: true,
    });
    Can i define the policies in code and to refer it to the bucket somehow?
    Thanks

**Comment by GaryAustin1** [comment, marked as answer]
You set policies for that call on the storage.buckets table.  You would need to add an insert policy for the user you want to create the bucket.

After you create the bucket, then to use it you will have to have RLS policies on storage.objects that handles all buckets or specific buckets.

You can set policies in the UI or with standard alter table sql to add policies
![image](https://github.com/supabase/supabase/assets/54564956/8963c99c-5fa8-448e-94d6-537cbe48c2fe)

**Reply by danielxpander** [reply]
Thank you @GaryAustin1  adding here relevant docs I've missed as well - 
https://supabase.com/docs/reference/javascript/storage-createbucket
https://supabase.com/docs/guides/storage/security/access-control
